src/SearchResults/index.js

```diff
@@ -668,7 +668,33 @@ function extractNormalizedFacetValues(results, attribute) {
       };
     });
   } else if (results._state.isHierarchicalFacet(attribute)) {
-    return find(results.hierarchicalFacets, predicate);
+    var hierarchicalFacetValues = find(results.hierarchicalFacets, predicate);
+    if (!hierarchicalFacetValues) return hierarchicalFacetValues;
+
+    var hierarchicalFacet = results._state.getHierarchicalFacetByName(attribute);
+    var currentRefinementSplit = unescapeFacetValue(
+      results._state.getHierarchicalRefinement(attribute)[0] || ''
+    ).split(results._state._getHierarchicalFacetSeparator(hierarchicalFacet));
+    currentRefinementSplit.unshift(attribute);
+
+    setIsRefined(hierarchicalFacetValues, currentRefinementSplit, 0);
```

Comment:
this option is mutating, as recreating the hierarchical tree requires information that is no longer available at this point using generateHierarchicalTree as far as I can tell